rpc/lib/client/http_client.go

```diff
@@ -74,6 +74,7 @@ func makeHTTPClient(remoteAddr string) (string, *http.Client) {
 	protocol, address, dialer := makeHTTPDialer(remoteAddr)
 	return protocol + "://" + address, &http.Client{
 		Transport: &http.Transport{
+			DisableCompression: true,
```

Comment:
We should add a comment above explaining why we set DisableCompression to true:

```suggestion
			// Set to true to prevent GZIP-bomb DoS attacks
			DisableCompression: true,
```